Utilities/stack_trace.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+#include <util/types.hpp>
+#include <util/logs.hpp>
+
+namespace utils
+{
+	std::vector<void*> backtrace(int max_depth = 255);
+	std::vector<std::string> backtrace_symbols(const std::vector<void*>& stack);
+
+	FORCE_INLINE void print_trace(logs::channel& logger, int max_depth = 255)

```

Comment:
The problem with force inline is that it is allowed to be ignored in some cases, it's actually safer to do the opposite: force function then remove it from the top of the backtrace,